WordPress/src/main/java/org/wordpress/android/ui/uploads/VideoOptimizerBase.java

```diff
@@ -0,0 +1,141 @@
+            return false;
+        }
+
+        if (mCacheDir == null) {
+            AppLog.w(AppLog.T.MEDIA, "VideoOptimizer > null cache dir");
+            mListener.onVideoOptimizationCompleted(mMedia);
+            return false;
+        }
+
+        if (!mCacheDir.exists() && !mCacheDir.mkdirs()) {
+            AppLog.w(AppLog.T.MEDIA, "VideoOptimizer > cannot create cache dir");
+            mListener.onVideoOptimizationCompleted(mMedia);
+            return false;
+        }
+
+        mOutputPath = mCacheDir.getPath() + "/" + mFilename;
+
+        return true;
+    }
+
+    protected void trackVideoProcessingEvents(boolean isError, Exception exception) {
+        Map<String, Object> properties = new HashMap<>();
+        Map<String, Object> inputVideoProperties =
+                AnalyticsUtils.getMediaProperties(getContext(), true, null, mInputPath);
+        putAllWithPrefix("input_video_", inputVideoProperties, properties);
+        if (mOutputPath != null) {
+            Map<String, Object> outputVideoProperties = AnalyticsUtils.getMediaProperties(
+                    getContext(),
+                    true,
+                    null,
+                    mOutputPath
+            );
+            putAllWithPrefix("output_video_", outputVideoProperties, properties);
+            String savedMegabytes =
+                    String.valueOf((FileUtils.length(mInputPath) - FileUtils.length(mOutputPath)) / (1024 * 1024));
+            properties.put("saved_megabytes", savedMegabytes);
+        }
+
+        long endTime = System.currentTimeMillis();
+        properties.put("elapsed_time_ms", endTime - mStartTimeMS);
+        if (isError) {
+            properties.put("exception_name", exception.getClass().getCanonicalName());
+            properties.put("exception_message", exception.getMessage());
+            AppLog.e(T.MEDIA, exception);
+        }
+        properties.put("optimizer-lib", "mp4composer");
+
+        AnalyticsTracker.Stat currentStatToTrack = isError ? MEDIA_VIDEO_OPTIMIZE_ERROR : MEDIA_VIDEO_OPTIMIZED;
+        AnalyticsTracker.track(currentStatToTrack, properties);
+    }
+
+    private void putAllWithPrefix(String prefix, Map<String, Object> inputMap, Map<String, Object> targetMap) {
+        if (inputMap != null && targetMap != null) {
+            for (Map.Entry<String, Object> entry : inputMap.entrySet()) {
+                targetMap.put(prefix + entry.getKey(), entry.getValue());
+            }
+        }
+    }
+
+    protected void selectMediaAndSendCompletionToListener() {
+        long originalFileSize = FileUtils.length(mInputPath);
+        long optimizedFileSize = FileUtils.length(mOutputPath);
+        long savings = originalFileSize - optimizedFileSize;
+
+        double savingsKb = Math.abs(savings) / 1024;
+        String strSavingsKb = new DecimalFormat("0.00").format(savingsKb).concat("KB");
+
+        // make sure the resulting file is smaller than the original
+        if (savings <= 0) {
```

Comment:
Would it be nice to delete the produced file in `mOutputPath`, once we realize we're not going to be using it after all?

Reply:
Good point 👍 ! if you agree I created a dedicated issue for this [here](https://github.com/wordpress-mobile/WordPress-Android/issues/14687)